### PR TITLE
[Refactor] Renamed ti.Index to ti.Axis

### PR DESCRIPTION
**Pull request by cruedo**
Related issue = #2516 
I'm a beginner, and is my first PR.
So pls let me know if I could have improved something in any way.
and apologies if I messed something up.
Thanks.

### CODE REVIEW

**Review by ljcc0930 — APPROVED** [review]
Good work!

File: taichi/ir/snode.h [hunk 1]
```diff
@@ -154,61 +154,60 @@ class SNode {
 
   SNode &insert_children(SNodeType t);
 
-  SNode &create_node(std::vector<Index> indices,
+  SNode &create_node(std::vector<Axis> indices,
```
Comment (k-ye) [inline taichi/ir/snode.h:157]: 1. Could you also help change the param name from `indices` to `axes`? (Thanks @strongoier for pointing this out)
2. https://github.com/taichi-dev/taichi/blob/f276de47518d900282ccaede5c1fe0a6f6170087/taichi/ir/snode.h#L32-L35 IndexExtractor also needs to be renamed to AxisExtractor

**Review by k-ye — COMMENTED** [review]
@strongoier this could cause some conflict with #2541, maybe it's better to wait for yours to land  first? :-)

**Comment by cruedo** [comment]
fixed them. 😄

**Comment by k-ye** [comment]
There's a merge conflict, ow this is good to go. Thx!

**Comment by cruedo** [comment]
Resolved : )

**Comment by cruedo** [comment]
Oh, it got merged.
Thank You ! @k-ye @ljcc0930 @strongoier and all others for your support ,
and giving me the opportunity to contribute :D

**Pull request merged**